### PR TITLE
Add Stackelberg Game to GUI

### PR DESCRIPTION
**Pull request by jgaupp**

### CODE REVIEW

File: src/gui/org/deidentifier/arx/gui/model/ModelStackelbergPrivacyCriterion.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+	private static final long serialVersionUID = -3632131196984677525L;
+	
+    private double             publisherBenefit  = 0d;
+    private double             publisherLoss     = 0d;
+    private double             adversaryGain     = 0d;
+    private double             adversaryCost     = 0d;
+
+	/** Parameter for the journalist model */
+    private DataSubset         subset            = null;
```
Comment (jgaupp) [inline src/gui/org/deidentifier/arx/gui/model/ModelStackelbergPrivacyCriterion.java]: DataSubset may not be necessary.  Included for continuity with framework representation.

File: src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		super(parent, model);
+	}
+
+	@Override
+	protected Composite build(Composite parent) {
+        // Create input group
+        Composite group = new Composite(parent, SWT.NONE);
+        group.setLayoutData(SWTUtil.createFillHorizontallyGridData());
+        GridLayout groupInputGridLayout = new GridLayout();
+        groupInputGridLayout.numColumns = 6;
```
Comment (jgaupp) [inline src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java]: Criteria editor is set up as two rows of six columns. versus one row for all other models.  When switching between models, the second row isn't automatically rendered, manual resize is necessary.  Need to redraw or change layout.

File: src/gui/org/deidentifier/arx/gui/resources/messages.properties [hunk 3]
```diff
@@ -778,6 +779,10 @@ CriterionDefinitionView.93=Delta:
 CriterionDefinitionView.94=Generalization:
 CriterionDefinitionView.100=D:
 CriterionDefinitionView.101=Significance level:
+CriterionDefinitionView.110=Publisher's Benefit:
+CriterionDefinitionView.111=Publisher's Loss:
+CriterionDefinitionView.112=Adverary's Gain:
+CriterionDefinitionView.113=Adverary's Cost:
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/resources/messages.properties]: For the sake of consistency, please don't use title case here.

File: src/gui/org/deidentifier/arx/gui/resources/messages.properties [hunk 4]
```diff
@@ -778,6 +779,10 @@ CriterionDefinitionView.93=Delta:
 CriterionDefinitionView.94=Generalization:
 CriterionDefinitionView.100=D:
 CriterionDefinitionView.101=Significance level:
+CriterionDefinitionView.110=Publisher's Benefit:
+CriterionDefinitionView.111=Publisher's Loss:
+CriterionDefinitionView.112=Adverary's Gain:
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/resources/messages.properties]: For the sake of consistency, please don't use title case here.

File: src/gui/org/deidentifier/arx/gui/resources/messages.properties [hunk 5]
```diff
@@ -778,6 +779,10 @@ CriterionDefinitionView.93=Delta:
 CriterionDefinitionView.94=Generalization:
 CriterionDefinitionView.100=D:
 CriterionDefinitionView.101=Significance level:
+CriterionDefinitionView.110=Publisher's Benefit:
+CriterionDefinitionView.111=Publisher's Loss:
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/resources/messages.properties]: For the sake of consistency, please don't use title case here.

File: src/gui/org/deidentifier/arx/gui/resources/messages.properties [hunk 6]
```diff
@@ -778,6 +779,10 @@ CriterionDefinitionView.93=Delta:
 CriterionDefinitionView.94=Generalization:
 CriterionDefinitionView.100=D:
 CriterionDefinitionView.101=Significance level:
+CriterionDefinitionView.110=Publisher's Benefit:
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/resources/messages.properties]: For the sake of consistency, please don't use title case here.

File: src/gui/org/deidentifier/arx/gui/resources/messages.properties [hunk 7]
```diff
@@ -400,6 +400,7 @@ ModelCriterion.0=k-Anonymity
 ModelCriterion.1=-Anonymity
 ModelCriterion.2=-Differential privacy
 ModelCriterion.3=(e,d)-Differential privacy
+ModelCriterion.4=Stackelberg Game Theory Privacy
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/resources/messages.properties]: For the sake of consistency, please don't use title case here.

File: src/gui/org/deidentifier/arx/gui/view/impl/define/ViewCriteriaList.java [hunk 8]
```diff
@@ -107,6 +109,7 @@ public ViewCriteriaList(final Composite parent, final Controller controller, Lay
         this.symbolD = controller.getResources().getManagedImage("symbol_d.png"); //$NON-NLS-1$
         this.symbolDP = controller.getResources().getManagedImage("symbol_dp.png"); //$NON-NLS-1$
         this.symbolR = controller.getResources().getManagedImage("symbol_r.png"); //$NON-NLS-1$
+        this.symbolG = controller.getResources().getManagedImage("symbol_g.png"); //$NON-NLS-1$
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/view/impl/define/ViewCriteriaList.java]: You even created a symbol! Great! +1

File: src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogCriterionUpdate.java [hunk 9]
```diff
@@ -264,6 +268,7 @@ protected Control createDialogArea(final Composite parent) {
         Image symbolD = controller.getResources().getManagedImage("symbol_d.png"); //$NON-NLS-1$
         Image symbolR = controller.getResources().getManagedImage("symbol_r.png"); //$NON-NLS-1$
         Image symbolDP = controller.getResources().getManagedImage("symbol_dp.png"); //$NON-NLS-1$
+        Image symbolG = controller.getResources().getManagedImage("symbol_g.png"); //$NON-NLS-1$
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogCriterionUpdate.java]: You even created a symbol! Great! :+1:

File: src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java [hunk 10]
```diff
@@ -0,0 +1,122 @@
+package org.deidentifier.arx.gui.view.impl.menu;
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java]: Please add the license header (see other files for reference)

File: src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+import de.linearbits.swt.widgets.Knob;
+
```
Comment (prasser) [inline src/gui/org/deidentifier/arx/gui/view/impl/menu/EditorCriterionStackelbergPrivacy.java]: Javadoc comments for members, methods and classes would be great.

**Comment by prasser** [comment]
This PR looks great! Thanks. I will need a little more time until I can pull the code and work on it, so I just added a few comments in the hope that they are helpful.

**Pull request merged**